### DISCUSSION TITLE
Should we have hashes in our asset file names?

### DISCUSSION THREAD

**Discussion by ampersarnie** [bigbite/build-tools#3] [General]
It's briefly mentioned in the [current `README.md`](https://github.com/bigbite/build-tools#getting-your-assets) that assets have hashes appended to them. I may need to get into this a little more (quick and dry is this was implemented as [VIP don't recommend `filemtime()` for version cache busting](https://docs.wpvip.com/technical-references/vip-platform/file-concatenation-and-minification/#2-versioning-to-bust-the-cache)), but I was wondering if anyone had any thoughts on this. It may be a non-issue if we implement [a config as discussed here](https://github.com/bigbite/build-tools/discussions/1) that allows for removal.

**Comment by jaymcp** [comment]
i haven't found it necessary to cache bust in this way in a long time. i set my asset version numbers from the theme/plugin (shown below), and haven't experienced issues with caching (including on sites running behind cloudflare, on which assets have year-long cache timeouts).

```php
// theme
add_action( 'wp_enqueue_scripts', function (): void {
    $theme = wp_get_theme();
    wp_enqueue_script( 'theme', 'https://domain.tld/path/to/file.js', [], $theme->get( 'Version' ), true );
} );

// plugin
add_action( 'wp_enqueue_scripts', function (): void {
    $plugin = get_plugin_data( __FILE__, false, false );
    wp_enqueue_script( 'plugin', plugins_url( '/path/to/file.js', __FILE__ ), [], $plugin['Version'], true );
} );
```

**Reply by ampersarnie** [reply]
Absolutely, that would have been the preferred method in the project that this was derived from (I can discuss specifics about that outside of here, incase the repo goes public), but it just wasn't possible. Having it as an option, and being defaulted to `off`, might make it helpful for future cases.
